### ISSUE TITLE
Provide user access to age of data files in reports

### ISSUE THREAD

**Issue by twelch** [seasketch/geoprocessing#125]
Need: as a project admin/coordinator, need to know what the age of the data files (e.g. surveys) used are, to know if they need updating for example.

* Add a `debug` or similar hidden section in reports
* Display list of datasources read and their last modified timestamp

**Comment by underbluewaters** [comment]
The service metadata includes a `.published` property that I could expose in the SSN admin interface. Data sources could be published separately though, right? There is some convoluted stuff going on to [expose a list of vector data sources](https://github.com/seasketch/geoprocessing/blob/87761717e055f2d1008910dd339d9dcc54d463ef/packages/geoprocessing/scripts/build/generateManifest.ts#L48). Maybe SSN could follow through somehow with that to check publishing times. We'd need something analogous for raster.

**Comment by twelch** [comment]
@underbluewaters  Yes datasources are published to the datasets buckets using `import:data` and `reimport:data` separate from cdk deployments.

If the services metadata root handler function can get access to the datasources.json file, then I think there's a couple options for populating that published field:
* datasources.json does include a last_updated timestamp for every datasource.  This could be read by the root handler serving up the service metadata.  The downside is this timestamp is not guaranteed to be accurate, it's updated by the `import:data` and `reimport:data` commands on the client side.  Someone could always implement code or manually run awscli commands to make changes to the S3 bucket.

* The root handler for the service could uses the datasources.json file to then lookup the last_modified timestamp in S3.  Other external datasources such as global (subdivided or other) would need to have their own published timestamps exposed.

**Comment by twelch** [comment]
The workaround for this feature, is just looking up the last modified timestamp manually in the S3 admin console.

**Comment by avmey** [comment]
s3 maintains a `Last-Modified` timestamp that is returned when data is fetched, so age of datasources used in reports can be found at runtime. 

The UI to start will be a small gear icon at the bottom of reports, which will have an option to view data timestamps (a JSON file structured like `{ ous_cultural_use: Fri, 19 Jan 2024 20:38:21 GMT, ... }`).

**Comment by twelch** [comment]
@avmey can you share a little more about how the timestamp is accessed?  Do you have to actually fetch some data to get the timestamp?
